spoon.rb

```diff
@@ -0,0 +1,57 @@
+STDOUT.sync = true # DO NOT REMOVE
+# Solution for https://www.codingame.com/ide/puzzle/there-is-no-spoon-episode-1
+class Spoon
+  attr_accessor :width, :height, :nodes
+
+  def initialize
+    @width = gets.to_i # the number of cells on the X axis
+    @height = gets.to_i # the number of cells on the Y axis
+    @nodes = []
+    init_nodes
+  end
+
+  def init_nodes
+    n = 0
+    height.times do
+      line = gets.chomp # width characters, each either 0 or .
+      find_node(line, n)
+      n += 1
+    end
+  end
+
+  def find_node(chars, n)
+    chars.each_char.with_index do |char, i|
+      next if char == '.'
+
+      nodes.push([i, n])
+    end
+  end
+
+  def find_right_node(node)
+    nodes.select { |n| n if n[1] == node[1] && n[0] > node[0] }.min
+  end
+
+  def find_bottom_node(node)
+    nodes.select { |n| n if n[0] == node[0] && n[1] > node[1] }.min
+  end
+
+  def add_coordinates(result)
```

Comment:
Do not rely on `add_coordinates` to add the whitespace. Think of a way to take that logic out of that method, so that the method only returns the pair of coordinates without whitespace